enhancements/builds/volume-mounted-resources.md

```diff
@@ -0,0 +1,391 @@
+  // +optional
+  ConfigMap *kapi.ConfigMapVolumeSource
+}
+```
+
+```go
+// BuildVolumeMount describes a mounting of a Volume within buildah's runtime environment.
+type BuildVolumeMount struct {
+  // This must match the Name of a BuildVolume, and cannot collide with volumes that are always added to the build pod by the build controller.
+  Name string
+  // Mounted read-only if true, read-write otherwise (false or unspecified).
+  // Defaults to false.
+  // +optional
+  ReadOnly bool
+  // Path within the build runtime environment at which the volume should be mounted.
+  // Must not contain ':', and cannot collide with a destination path generated by the builder process.
+  MountPath string
+}
+```
+
+Example usage (use of the existing secret/configmap injection api is included for comparison, it is not changing):
+
+```yaml
+apiVersion: v1
+items:
+- apiVersion: build.openshift.io/v1
+  kind: BuildConfig
+  metadata:
+    name: mybuild
+    namespace: p1
+  spec:
+    strategy:
+      sourceStrategy:
+        from:
+          kind: ImageStreamTag
+          name: nodejs:10-SCL
+          namespace: openshift
+        volumes:
+        - name: secret
+          secret:
+            secretName: somesecret
+        - name: config
+          configMap:
+            name: someconfigmap
+            items:
+            - key: somekey
+              path: volume/path/value.txt
+        - name: content-access
+          csi:
+            driver: csi-driver-projected-resource.openshift.io
+            volumeAttributes:
+              share: etc-pki-entitlement
+        volumeMounts:
+        - name: config
+          mountPath: /tmp/config
+        - name: secret
+          mountPath: /tmp/secret
+        - name: content-access
+          mountPath: /etc/pki/entitlement
+      type: Source
+    source:
+      secrets:
+      - secret: 
+          name: myOtherSecret
+        destinationDir: /tmp/othersecret
+      configMaps:
+      - configMap:
+          name: myOtherConfigMap
+        destinationDir: /tmp/otherconfig
+```
+
+#### Volume Name and Destination Collisions
+
+Builds today have several container volume mounts that are provided by the build controller, including:
+
+- Node pull secrets (host)
+- System configuration (ConfigMap)
+- Certificate authorities for the cluster proxy and internal registry (ConfigMap)
+- Blob metadata cache (host)
+- RHEL entitlements (host via cri-o configuration)
+
+In addition, the builder process generates the following buildah volume mounts:
+
+- RHEL entitlements (destination is `/run/secrets/etc-pki-entitlement`, `/run/secrets/redhat.repo`, and `/run/secrets/rhsm`)
+- Custom PKI trust bundle - destination is `/etc/pki/ca-trust`, added via the Build/BuildConfig's `mountTrustedCA` option.
+
+If an input volume's name collides with a volume created by the build controller, or implied by OpenShift's cri-o configuration, the build should fail to run.
```

Comment:
we should be able to generate our own volume names(both on the pod and in buildah) such that we avoid collisions.  We don't have to use the volume name the user provides, when setting up volumes.

Reply:
True about creating our own volume names, though we get into fun situations where we need to validate/truncate superficially long volume names.

Reply:
@bparees I made this a "we can change this in a future enhancement" given the short window we have to land this in 4.8

Reply:
btw i wasn't suggesting random suffixes.  you don't have to use any part of the user given name, you can just call it "vol1", "vol2", etc.  So no issue w/ truncation.  Again we already do this when mounting the secrets/configmaps into the build pod.  (generate our own volume names)